server/core/src/miscServerFunct.cpp

```diff
@@ -2976,53 +2983,12 @@ irods::error get_script_output_single_line(
     const std::string&              script_name,
     const std::vector<std::string>& args,
     std::string&                    output ) {
+    bp::ipstream output_stream;
+    bp::child c(script_language, args, bp::std_out > output_stream);
     output.clear();
-    std::stringstream exec;
-    try {
-    exec << script_language
-         << " " << irods::get_irods_home_directory().string()
-         << "/scripts/" << script_name;
-    } catch (const irods::exception& e) {
-        irods::log(e);
-        return ERROR(-1, "failed to get irods home directory");
-    }
-    for ( std::vector<std::string>::size_type i = 0; i < args.size(); ++i ) {
-        exec << " " << args[i];
-    }
-
-    FILE *fp = popen( exec.str().c_str(), "r" );
-    if ( fp == NULL ) {
-        return ERROR( SYS_FORK_ERROR, "popen() failed" );
-    }
-
-    std::vector<char> buf( 1000 );
-    const char* fgets_ret = fgets( &buf[0], buf.size(), fp );
-    if ( fgets_ret == NULL ) {
-        std::stringstream msg;
-        msg << "fgets() failed. feof["
-            << std::feof( fp )
-            << "] ferror["
-            << std::ferror( fp ) << "]";
-        const int pclose_ret = pclose( fp );
-        msg << " pclose[" << pclose_ret << "]";
-        return ERROR( FILE_READ_ERR, msg.str() );
-    }
-
-    const int pclose_ret = pclose( fp );
-    if ( pclose_ret == -1 ) {
-        return ERROR( SYS_FORK_ERROR,
-                      "pclose() failed." );
-    }
-
-    output = &buf[0];
-    // Remove trailing newline
-    const std::string::size_type size = output.size();
-    if ( size > 0 && output[size - 1] == '\n' ) {
-        output.resize( size - 1 );
-    }
-
+    std::getline(output_stream, output);
```

Comment:
The examples indicate we should check `output_stream` before using it, which makes sense.

Reply:
what do you mean by checking it?

Reply:
We check the stream to make sure it isn't in a bad state.
```cpp
if (!output_stream) {
    // Stream is in a bad state.
    // Handle error.
}

std::getline(output_stream, output);
```